consensus/src/execution_pipeline.rs

```diff
@@ -173,8 +179,10 @@ impl ExecutionPipeline {
             parent_block_id,
             block_executor_onchain_config,
             result_tx,
+            command_creation_time,
         }) = block_rx.recv().await
         {
+            counters::EXECUTE_BLOCK_WAIT_TIME.observe_duration(command_creation_time.elapsed());
```

Comment:
I would prefer if we can be consistent around these metrics and use `observer_block` API to denote different stages. WDYT?

Reply:
When a block enters a certain stage, we are using the `observe_block` API to measure the time elapsed from block creation.

Here, I'm basically measuring the wait time from the time the block is prepared to the time block execution started. This usage is different from `observe_block` usage?

Reply:
Yes, you can still do that with observe_block API with something like `before_prepare` and `prepared` stage right?

Reply:
Before this PR, I was trying that way of subtracting observe_block API output for `executed` and `ordered` stages in [grafana](https://grafana.aptoslabs.com/explore?schemaVersion=1&panes=%7B%22so1%22%3A%7B%22datasource%22%3A%22vU-Lwva4k%22%2C%22queries%22%3A%5B%7B%22datasource%22%3A%7B%22type%22%3A%22prometheus%22%2C%22uid%22%3A%22vU-Lwva4k%22%7D%2C%22editorMode%22%3A%22code%22%2C%22expr%22%3A%22rate%28consensus_duration_sum%7Bchain_name%3D%7E%5C%22mainnet%5C%22%2C+cluster%3D%7E%5C%22.*%5C%22%2C+metrics_source%3D%7E%5C%22telemetry-service%5C%22%2C+namespace%3D%7E%5C%22.*%5C%22%2C+op%3D%27execute_block%27%2C+kubernetes_pod_name%3D%7E%5C%22.*%5C%22%2C+role%3D%7E%5C%22%28aws-load-balancer-controller%7Cdb_backup%7Cdb_backup_compaction%7Cdb_backup_verify%7Cendpoints%7Cfullnode%7Cfullnode-archival%7Chaproxy%7Cnode%7Cpod%7Cprometheus-node-exporter%7Cpublic_fullnode%7Cservice%7Cvalidator%7Cvalidator_fullnode%7Cvector%29%5C%22%7D%5B1m%5D%29+%2F+rate%28consensus_duration_count%7Bchain_name%3D%7E%5C%22mainnet%5C%22%2C+cluster%3D%7E%5C%22.*%5C%22%2C+metrics_source%3D%7E%5C%22telemetry-service%5C%22%2C+namespace%3D%7E%5C%22.*%5C%22%2C+op%3D%27execute_block%27%2C+kubernetes_pod_name%3D%7E%5C%22.*%5C%22%2C+role%3D%7E%5C%22%28aws-load-balancer-controller%7Cdb_backup%7Cdb_backup_compaction%7Cdb_backup_verify%7Cendpoints%7Cfullnode%7Cfullnode-archival%7Chaproxy%7Cnode%7Cpod%7Cprometheus-node-exporter%7Cpublic_fullnode%7Cservice%7Cvalidator%7Cvalidator_fullnode%7Cvector%29%5C%22%7D%5B1m%5D%29+%5Cn%5Cn-+%5Cn%5Cnrate%28consensus_duration_sum%7Bchain_name%3D%7E%5C%22mainnet%5C%22%2C+cluster%3D%7E%5C%22.*%5C%22%2C+metrics_source%3D%7E%5C%22telemetry-service%5C%22%2C+namespace%3D%7E%5C%22.*%5C%22%2C+op%3D%27ordered%27%2C+kubernetes_pod_name%3D%7E%5C%22.*%5C%22%2C+role%3D%7E%5C%22%28aws-load-balancer-controller%7Cdb_backup%7Cdb_backup_compaction%7Cdb_backup_verify%7Cendpoints%7Cfullnode%7Cfullnode-archival%7Chaproxy%7Cnode%7Cpod%7Cprometheus-node-exporter%7Cpublic_fullnode%7Cservice%7Cvalidator%7Cvalidator_fullnode%7Cvector%29%5C%22%7D%5B1m%5D%29+%2F+rate%28consensus_duration_count%7Bchain_name%3D%7E%5C%22mainnet%5C%22%2C+cluster%3D%7E%5C%22.*%5C%22%2C+metrics_source%3D%7E%5C%22telemetry-service%5C%22%2C+namespace%3D%7E%5C%22.*%5C%22%2C+op%3D%27ordered%27%2C+kubernetes_pod_name%3D%7E%5C%22.*%5C%22%2C+role%3D%7E%5C%22%28aws-load-balancer-controller%7Cdb_backup%7Cdb_backup_compaction%7Cdb_backup_verify%7Cendpoints%7Cfullnode%7Cfullnode-archival%7Chaproxy%7Cnode%7Cpod%7Cprometheus-node-exporter%7Cpublic_fullnode%7Cservice%7Cvalidator%7Cvalidator_fullnode%7Cvector%29%5C%22%7D%5B1m%5D%29%22%2C%22format%22%3A%22time_series%22%2C%22legendFormat%22%3A%22%7B%7Bkubernetes_pod_name%7D%7D-%7B%7Brole%7D%7D%22%2C%22range%22%3Atrue%2C%22refId%22%3A%22A%22%2C%22interval%22%3A%22%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%221724443380178%22%2C%22to%22%3A%221724443726737%22%7D%7D%7D&orgId=1). It's giving me this no data graph. Subtracting two quantiles of observe_block API works. But subtracting observe_block API for individual validators isn't working.

Reply:
Before this PR, I was trying that way of subtracting observe_block API output for `executed` and `ordered` stages in grafana. It's giving me this no data graph. Subtracting two quantiles of observe_block API works. But subtracting observe_block API for individual validators isn't working. So, had to implemented this way.

https://grafana.aptoslabs.com/goto/udLbUOqIR?orgId=1